libcaf_test/caf/test/runnable.hpp

```diff
@@ -65,6 +65,9 @@ class CAF_TEST_EXPORT runnable {
                 const detail::source_location& location
                 = detail::source_location::current()) {
     assert_save_comparison<T0, T1>();
+    static_assert(
+      !(std::is_floating_point_v<T0> && std::is_floating_point_v<T1>),
+      "comparing floating points using operator== is unsafe, use approx");
```

Comment:
That same assert should go into `require_eq`.

Reply:
As `require_eq` calls `check_eq` internally, the static_assert will be triggered when trying to compare floating point numbers with `require_eq(res, 1.0f)`. Should we duplicate the static_assert anyways?

Reply:
Nah, sorry for not double-checking. 👍